streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -530,6 +530,14 @@ public void shouldSetInternalLeaveGroupOnCloseConfigToFalseInConsumer() {
         assertThat(consumerConfigs.get("internal.leave.group.on.close"), is(false));
     }
 
+    @Test
+    public void shouldResetToDefaultIfConsumerAllowAutoCreateTopicsIsOverridden() {
```

Comment:
This should apply to all consumers, right? Should we extend the test accordingly?

Should we also capture the logs and verify that the WARN is printed (not sure if necessary)?

Reply:
Yes, will be making this test extensible for all the consumers. 
Can you help me with how can I capture the log to verify WARN is being printed? I think it's a good-to-have test case. Any test that already does this?

Reply:
> Can you help me with how can I capture the log to verify WARN is being printed? I

There is `org.apache.kafka.common.utils.LogCaptureAppender` class -- it's used in other tests, too, that you can use a blue print.